### PR TITLE
Denraf/us 24823/helmchart oom event generator

### PR DESCRIPTION
**Pull request by denraf**
**Why This PR?**
Adding a helm chart to install [Kubernetes-oom-event-generator](https://github.com/xing/kubernetes-oom-event-generator)

Fixes #

**Changes**
Changes proposed in this pull request:

*
*

**Checklist:**

* [x] I have updated the chart version in `Chart.yaml` following Semantic Versioning.
* [x] Any new values have been added to the README for the Chart, or helm-docs has been run for the charts that support it.

Generated template:
```
---
# Source: oom-event-generator/templates/serviceaccount.yaml
apiVersion: v1
kind: ServiceAccount
metadata:
  name: oom-event-generator
  labels:
    helm.sh/chart: oom-event-generator-0.1.0
    app.kubernetes.io/name: oom-event-generator
    app.kubernetes.io/instance: oom-event-generator
    app.kubernetes.io/version: "1.2.0"
    app.kubernetes.io/managed-by: Helm
---
# Source: oom-event-generator/templates/clusterrole.yaml
apiVersion: rbac.authorization.k8s.io/v1beta1
kind: ClusterRole
metadata:
  name: oom-event-generator
  labels:
    helm.sh/chart: oom-event-generator-0.1.0
    app.kubernetes.io/name: oom-event-generator
    app.kubernetes.io/instance: oom-event-generator
    app.kubernetes.io/version: "1.2.0"
    app.kubernetes.io/managed-by: Helm
rules:
  - apiGroups:
      - ""
    resources:
      - pods
      - pods/status
    verbs:
      - get
      - list
      - watch
  - apiGroups:
      - ""
    resources:
      - events
    verbs:
      - create
      - patch
      - list
      - watch
---
# Source: oom-event-generator/templates/clusterrolebinding.yaml
apiVersion: rbac.authorization.k8s.io/v1beta1
kind: ClusterRoleBinding
metadata:
  labels:
    helm.sh/chart: oom-event-generator-0.1.0
    app.kubernetes.io/name: oom-event-generator
    app.kubernetes.io/instance: oom-event-generator
    app.kubernetes.io/version: "1.2.0"
    app.kubernetes.io/managed-by: Helm
  name: oom-event-generator
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: oom-event-generator
subjects:
- kind: ServiceAccount
  name: oom-event-generator
  namespace: default
---
# Source: oom-event-generator/templates/deployment.yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: oom-event-generator
  labels:
    helm.sh/chart: oom-event-generator-0.1.0
    app.kubernetes.io/name: oom-event-generator
    app.kubernetes.io/instance: oom-event-generator
    app.kubernetes.io/version: "1.2.0"
    app.kubernetes.io/managed-by: Helm
spec:
  replicas: 1
  selector:
    matchLabels:
      app.kubernetes.io/name: oom-event-generator
      app.kubernetes.io/instance: oom-event-generator
  template:
    metadata:
      labels:
        app.kubernetes.io/name: oom-event-generator
        app.kubernetes.io/instance: oom-event-generator
    spec:
      serviceAccountName: oom-event-generator
      securityContext:
        {}
      containers:
        - name: oom-event-generator
          securityContext:
            {}
          image: "xingse/kubernetes-oom-event-generator:v1.2.0"
          imagePullPolicy: Always
          env:
          resources:
            limits:
              cpu: 100m
              memory: 128Mi
            requests:
              cpu: 100m
              memory: 128Mi
```

### CODE REVIEW

**Review by sudermanjr — CHANGES_REQUESTED** [review]
I recommend using helm-docs for the values table. That way it's easier to update. This repo supports it in CI as well.

Also, can you modify the CODEOWNERS file for this repo for this new chart?

**Comment by sudermanjr** [comment]
https://github.com/norwoodj/helm-docs

File: incubator/oom-event-generator/values.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+# Declare variables to be passed into your templates.
+
+# repliaCount -- The number of replicas to run
+replicaCount: 1
+
+image:
+  repository: xingse/kubernetes-oom-event-generator
+  pullPolicy: Always
+  # Overrides the image tag whose default is the chart appVersion.
+  tag: "v1.2.0"
```
Comment (sudermanjr) [inline incubator/oom-event-generator/values.yaml:12]: The standard helm chart that I think you used automatically grabs the `appVersion` from the chart.yaml, so I think you can leave this empty.

File: incubator/oom-event-generator/values.yaml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+securityContext: {}
+  # capabilities:
+  #   drop:
+  #   - ALL
+  # readOnlyRootFilesystem: true
+  # runAsNonRoot: true
+  # runAsUser: 1000
+
+env: {}
```
Comment (sudermanjr) [inline incubator/oom-event-generator/values.yaml:43]: Probably want a comment on this one.

File: CODEOWNERS [hunk 3]
```diff
@@ -12,6 +12,7 @@
 /incubator/opa/ @sudermanjr
 /incubator/ro-cert-manager/ @sudermanjr
 /incubator/route53sync/ @sudermanjr
+/incubator/oom-event-generator/ @sudermanjr
```
Comment (sudermanjr) [inline CODEOWNERS]: you put yourself as the maintainer in the Chart.yaml, so I was thinking more along the lines of:

```suggestion
/incubator/oom-event-generator/ @denraf 
```

**Review by sudermanjr — CHANGES_REQUESTED** [review]

File: incubator/oom-event-generator/values.yaml (hunk 1 above)
Comment (denraf) [reply incubator/oom-event-generator/values.yaml:12]: appVersion is `1.2.0`
tag is `v1.2.0` -- there is no tag `1.2.0` per docker hub

File: CODEOWNERS (hunk 3 above)
Comment (denraf) [reply CODEOWNERS]: Done

File: incubator/oom-event-generator/values.yaml (hunk 2 above)
Comment (denraf) [reply incubator/oom-event-generator/values.yaml:43]: Done

File: incubator/oom-event-generator/values.yaml (hunk 1 above)
Comment (sudermanjr) [reply incubator/oom-event-generator/values.yaml:12]: Why not set appVersion to the image tag then? That way we don't have to maintain both.

**Review by sudermanjr — APPROVED** [review]
I had another comment about the app version. It's not blocking for me though￼

**Pull request merged**